connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/mcp/discovery/McpClientGatewayToolHandler.java

```diff
@@ -50,13 +52,15 @@ public String type() {
     return GATEWAY_TYPE;
   }
 
+  @Override
+  public boolean isGatewayManaged(String toolName) {
+    return McpToolCallIdentifier.isMcpToolCallIdentifier(toolName);
+  }
```

Comment:
Maybe we would have the possibility to not rely on the naming convention for these lookups by adding some metadata to the tool definitions upon resolving (e.g. which gateway it is managing it). I'll look into that in a follow-up.